src/wayland/generated/wayland_wrapper.cpp

```diff
@@ -191,6 +194,9 @@ mw::Compositor::Compositor(struct wl_resource* resource, Version<4>)
     wl_resource_set_implementation(resource, Thunks::request_vtable, this, &Thunks::resource_destroyed_thunk);
 }
 
+mw::Compositor::Compositor(struct wl_client* client, struct wl_resource* parent, Version<4> static_version)
+    : Compositor{wl_resource_create(client, &wl_compositor_interface_data , wl_resource_get_version(parent), 0), static_version} {}
```

Comment:
Why pass in a resource and a client? Why not use `wl_resource_get_client()`?

Reply:
There is an existing signature for passing the resource from the client request:
```c++
    Compositor(struct wl_resource* resource, Version<3>);
```

Reply:
So, I think that:
 * For globals, this constructor doesn't make sense; instances of a global can only be constructed from the `bind(wl_resource* new_resource)` method, which already has a resource
 * For other resources (eg: `mw::Surface`), I *think* this can quite easily be reworked as `Surface(Compositor* parent)`. And then we'd (maybe) remove the raw `wl_resource*` constructor(?)